packages/muelu/src/Misc/MueLu_MultiVectorAggregationTransferFactory_decl.hpp

```diff
@@ -0,0 +1,142 @@
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact
+//                    Jonathan Hu       (jhu@sandia.gov)
+//                    Andrey Prokopenko (aprokop@sandia.gov)
+//                    Ray Tuminaro      (rstumin@sandia.gov)
+//
+// ***********************************************************************
+//
+// @HEADER
+#ifndef MUELU_MULTIVECTORAGGREGATIONTRANSFER_FACTORY_DECL_HPP
+#define MUELU_MULTIVECTORAGGREGATIONTRANSFER_FACTORY_DECL_HPP
+
+#include "MueLu_ConfigDefs.hpp"
+#include "MueLu_TwoLevelFactoryBase.hpp"
+#include "Xpetra_MultiVector_fwd.hpp"
+#include "Xpetra_MultiVectorFactory_fwd.hpp"
+#include "Xpetra_Matrix.hpp"
+
+#include "MueLu_CoarseMapFactory_fwd.hpp"
+#include "MueLu_MultiVectorAggregationTransferFactory_fwd.hpp"
+#include "MueLu_Utilities_fwd.hpp"
+
+namespace MueLu {
+
+/*!
+  @class MultiVectorAggregationTransferFactory class.
+  @brief Class for transferring a MultiVector from a finer level to a coarser one using aggregates
+
+    This is separate from MultiVectorTransferFactory and CoordinatesTransferFactory
+    as it transfers non-coordinate quantities based only on aggregate information.
+
+  ## Input/output of MultiVectorAggregationTransferFactory ##
+
+  ### User parameters of MultiVectorAggregationTransferFactory ###
+  Parameter | type | default | master.xml | validated | requested | description
+  ----------|------|---------|:----------:|:---------:|:---------:|------------
+  | Vector name| string | "undefined" |   | * | (*) | Name of vector
+  | Vector factory | Factory | null |   | * | (*) | Factory providing vector
```

Comment:
`DeclarInput()` also declares `Aggregates` and `CoarseMap` as input. Shouldn't they show up in this table as well?